fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/QueryStatsAction.java

```diff
@@ -0,0 +1,292 @@
+ */
+@RestController
+public class QueryStatsAction extends RestBaseController {
+    private static final Logger LOG = LogManager.getLogger(QueryStatsAction.class);
+
+    @RequestMapping(path = "/api/query_stats/{catalog}", method = RequestMethod.GET)
+    protected Object getQueryStatsFromCatalog(@PathVariable("catalog") String catalog,
+            @RequestParam(name = "summary", required = false, defaultValue = "true") boolean summary,
+            @RequestParam(name = "pretty", required = false, defaultValue = "false") boolean pretty,
+            HttpServletRequest request, HttpServletResponse response) {
+        if (pretty && summary) {
+            return ResponseEntityBuilder.badRequest("pretty and summary can not be true at the same time");
+        }
+        executeCheckPassword(request, response);
+        checkGlobalAuth(ConnectContext.get().getCurrentUserIdentity(), PrivPredicate.ADMIN);
+        // use NS_KEY as catalog, but NS_KEY's default value is 'default_cluster'.
+        if (catalog.equalsIgnoreCase(SystemInfoService.DEFAULT_CLUSTER)) {
+            catalog = InternalCatalog.INTERNAL_CATALOG_NAME;
+        }
+        try {
+            Map<String, Map> result = Env.getCurrentEnv().getQueryStats().getStats(catalog, summary);
+            if (pretty) {
+                return ResponseEntityBuilder.ok(getPrettyJson(result.get("detail"), QueryStatsType.DATABASE));
+            }
+            return ResponseEntityBuilder.ok(result);
+        } catch (Exception e) {
+            LOG.warn("get query stats from catalog {} failed", catalog, e);
+            return ResponseEntityBuilder.internalError(e.getMessage());
+        }
+    }
+
+    @RequestMapping(path = "/api/query_stats/{catalog}/{database}", method = RequestMethod.GET)
+    protected Object getQueryStatsFromDatabase(@PathVariable("catalog") String catalog,
+            @PathVariable("database") String database,
+            @RequestParam(name = "summary", required = false, defaultValue = "true") boolean summary,
+            @RequestParam(name = "pretty", required = false, defaultValue = "false") boolean pretty,
+            HttpServletRequest request, HttpServletResponse response) {
+        if (pretty && summary) {
+            return ResponseEntityBuilder.badRequest("pretty and summary can not be true at the same time");
+        }
+        executeCheckPassword(request, response);
+        checkDbAuth(ConnectContext.get().getCurrentUserIdentity(), database, PrivPredicate.SHOW);
+        // use NS_KEY as catalog, but NS_KEY's default value is 'default_cluster'.
+        String clasterName = catalog;
+        if (catalog.equalsIgnoreCase(SystemInfoService.DEFAULT_CLUSTER)) {
+            catalog = InternalCatalog.INTERNAL_CATALOG_NAME;
+        }
+        database = ClusterNamespace.getFullName(clasterName, database);
+        try {
+            Map<String, Map> result = Env.getCurrentEnv().getQueryStats().getStats(catalog, database, summary);
```

Comment:
May be we should unify the action and show stmt in later pr.
The action should also get merged stat from each FE.